src/Integration.Vsix.UnitTests/Suppression/DelegateInjectorTests.cs

```diff
@@ -0,0 +1,287 @@
+                Assembly validAsm1 = CreateAndLoadAssembly(SonarAnalyzerAssemblyName, "105.0.1", SourceCodeWithStaticProperty);
+                AssertSuppressionPropertyIsSet(validAsm1);
+
+                Assembly withoutPropertyAsm = CreateAndLoadAssembly("withoutProperty", "105.0", "class Class1{}");
+
+                Assembly validAsm2 = CreateAndLoadAssembly(SonarAnalyzerAssemblyName, "105.0.2", SourceCodeWithStaticProperty);
+                AssertSuppressionPropertyIsSet(validAsm2);
+            };
+
+            // Injector has been disposed so should not set the value
+            Assembly validAsm3 = CreateAndLoadAssembly(SonarAnalyzerAssemblyName, "105.0.3", SourceCodeWithStaticProperty);
+            AssertSuppressionPropertyIsNotSet(validAsm3);
+        }
+
+        #region Assertion methods
+
+        private static void AssertSuppressionPropertyIsNotSet(Assembly asm)
+        {
+            object propertyValue = GetSuppressionPropertyValue(asm);
+            propertyValue.Should().BeNull();
+        }
+
+        private static void AssertSuppressionPropertyIsSet(Assembly asm)
+        {
+            object propertyValue = GetSuppressionPropertyValue(asm);
+            propertyValue.Should().NotBeNull();
+        }
+
+        private static void AssertSuppressionPropertyDoesNotExist(Assembly asm)
+        {
+            PropertyInfo propertyInfo = GetStaticProperty(asm);
+            propertyInfo.Should().BeNull("not expecting the static property to exist in this test assembly");
+        }
+
+        private static object GetSuppressionPropertyValue(Assembly asm)
+        {
+            PropertyInfo propertyInfo = GetStaticProperty(asm);
+            propertyInfo.Should().NotBeNull();
+
+            return propertyInfo.GetValue(null);
+        }
+
+        private static PropertyInfo GetStaticProperty(Assembly asm)
+        {
+            Type contextType = asm.GetTypes().SingleOrDefault(t => t.Name == "SonarAnalysisContext");
+            PropertyInfo propertyInfo = contextType?.GetProperty("ShouldDiagnosticBeReported", BindingFlags.Static | BindingFlags.Public);
+            return propertyInfo;
+        }
+
+        #endregion
+
+        #region Assembly creation
+
+        private Assembly CreateAndLoadAssembly(string assemblyName, string assemblyVersion, string source)
+        {
+            Compilation comp = CreateCompilation(assemblyName, assemblyVersion, source);
+
+            using (var stream = new MemoryStream())
+            {
+                EmitResult result = comp.Emit(stream);
+                if (!result.Success)
+                {
+                    this.TestContext.WriteLine("Test setup error: compilation of test assembly failed. Errors:");
+                    foreach (Diagnostic diag in result.Diagnostics)
+                    {
+                        this.TestContext.WriteLine(diag.ToString());
+                    }
+                }
+                result.Success.Should().BeTrue();
```

Comment:
I am not a big fan of assertions done during the _Arrange_ part but you can keep it this way. I would have thrown some exception (or maybe even `AggregateException`).

Reply:
Changed to an _Exception_